src/testdir/term_util.vim

```diff
@@ -24,6 +24,18 @@ func StopShellInTerminal(buf)
   call WaitFor({-> job_status(job) == "dead"})
 endfunc
 
+" Wrapper around term_wait() to allow more time for re-runs of flaky tests
+func TermWait(buf, ...)
+  let wait_time = a:0 ? a:1 : 10
+  if g:run_nr == 1
+    call term_wait(a:buf, wait_time)
+  elseif g:run_nr == 2
+    call term_wait(a:buf, wait_time * 4)
+  else
+    call term_wait(a:buf, wait_time * 10)
+  endif
+endfunc
```

Comment:
Personally, I don't understand well what `term_wait()` does.
Instead of using `term_wait()` in tests, shouldn't those test instead wait for an expected output in the terminal with something like this?
```
call WaitForAssert({-> assert_match('...pattern...', term_getline(buf, a_line_nr))}, 1000)
```

Reply:
Deterministic waits are *much* preferred to arbitrary delays.  However, since there are a lot of arbitrary delays in the tests and those sporadically cause failures in both CI and Debian builds, I went with a quick band-aid to help resolve the general problem.